pkg/backup/actions/csi/pvc_action.go

```diff
@@ -564,3 +572,65 @@ func NewPvcBackupItemAction(f client.Factory) plugincommon.HandlerInitializer {
 		}, nil
 	}
 }
+
+func setPVCRequestSizeToVSRestoreSize(
+	pvc *corev1api.PersistentVolumeClaim,
+	crClient crclient.Client,
+	volumeSnapshotName string,
+	logger logrus.FieldLogger,
+) error {
+	vs := new(snapshotv1api.VolumeSnapshot)
+	if err := crClient.Get(context.TODO(),
+		crclient.ObjectKey{
+			Namespace: pvc.Namespace,
+			Name:      volumeSnapshotName,
+		},
+		vs,
+	); err != nil {
+		return errors.Wrapf(err, "Failed to get Volumesnapshot %s/%s to restore PVC %s/%s",
+			pvc.Namespace, volumeSnapshotName, pvc.Namespace, pvc.Name)
+	}
+
+	if _, exists := vs.Annotations[velerov1api.VolumeSnapshotRestoreSize]; exists {
```

Comment:
This won't exist yet, since it's only set in the VS BIA (which will be called *after* this BIA exits):
https://github.com/vmware-tanzu/velero/blob/main/pkg/backup/actions/csi/volumesnapshot_action.go#L169

So instead of using the annotation, we will need to use `*vsc.Status.RestoreSize` (obtained by calling `vsc,csi.WaitUntilVSCHandleIsReady` -- this is already called elsewhere under certain circumstances, so we probably need to refactor the PVC BIA to always call it.